internal/webhook/template_webhook.go

```diff
@@ -185,11 +202,72 @@ func (*ProviderTemplateValidator) ValidateUpdate(_ context.Context, _, _ runtime
 }
 
 // ValidateDelete implements webhook.Validator so a webhook will be registered for the type.
-func (*ProviderTemplateValidator) ValidateDelete(_ context.Context, _ runtime.Object) (admission.Warnings, error) {
+func (v *ProviderTemplateValidator) ValidateDelete(ctx context.Context, obj runtime.Object) (admission.Warnings, error) {
+	template, ok := obj.(*v1alpha1.ProviderTemplate)
+	if !ok {
+		return admission.Warnings{"Wrong object"}, apierrors.NewBadRequest(fmt.Sprintf("expected ProviderTemplate but got a %T", obj))
+	}
+
+	owners := getOwnersWithKind(template, v1alpha1.ReleaseKind)
+	if len(owners) > 0 {
+		return admission.Warnings{fmt.Sprintf("The ProviderTemplate %s cannot be removed while it is part of existing Releases: %s",
+			template.GetName(), strings.Join(owners, ", "))}, errTemplateDeletionForbidden
+	}
+
+	mgmtList := &v1alpha1.ManagementList{}
```

Comment:
nit: we already have this logic in the release webhook, let's make a common function out of it